Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts

```diff
@@ -778,12 +778,16 @@ namespace gdjs {
         this._findPlatformHighestRelativeYUnderObject(platform, context);
         let highestRelativeY = context.getFloorDeltaY();
         if (
-          // When following the floor, ignore jumpthrus that are higher than the character bottom.
-          this._state === this._onFloor &&
-          platform !== this._onFloor.getFloorPlatform() &&
           platform.getPlatformType() ===
             gdjs.PlatformRuntimeBehavior.JUMPTHRU &&
-          highestRelativeY < 0
+          // When following the floor, ignore jumpthrus that are higher than the character bottom.
+          ((this._state === this._onFloor &&
+            platform !== this._onFloor.getFloorPlatform() &&
+            highestRelativeY < 0) ||
+            // A jumpthrus should never constrain a character to go below.
+            // Jumpthrus are considered as obstacles at the 1st frame they are overlapping the character
+            // because it allows it to land on them, but they shouldn't push on its head.
+            context.allowedMinDeltaY !== previousAllowedMinDeltaY)
```

Comment:
Make sense. Can you tell me what are the reasons that would make allowedMinDeltaY to change? (just for me to fully understand what could cause this)

Reply:
If a solid platform is on the top of the character, it can land to another platform if it's not squeezed between the 2.

If the obstacle were a solid platform the character would have been stopped before it when moving on X and it would not have been used for floor search. So this scenario can't happen with solid platforms unless it's a moving platform. In this case, the character is pushed in the ground and move back up when the moving platform goes away.